### PR TITLE
Meta tag names

### PR DESCRIPTION
**Pull request by AL-Hareth**
# Overview

# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

Changed the type of the name property on the DocumentMeta interface from `string` to union types that includes all the values possible in a name attribute inside the meta tag

from: 
```ts
interface DocumentMeta {
    // ...
    readonly name?: string;
    // ...
}
```
to:
```ts
interface DocumentMeta {
    // ...
    readonly name?: "application-name" | "author" | "keyword" /* ...etc */;
    // ...
}
```

# Use cases and why

This gives better development experience where you get the correct name values if you're trying to add a meta tag data using qwik-city

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by mhevery** [comment]
Are you still working on this change or is it abandoned?

**Comment by AL-Hareth** [comment]
> Are you still working on this change or is it abandoned?

I would be happy to work on it, but I don't really understand the process that I should go with before making the pull request

**Comment by mhevery** [comment]
I guess it just needs a rebase? Let me see if I can make it pass...

**Comment by mhevery** [comment]
It looks like this change has already been merged https://github.com/BuilderIO/qwik/pull/4340

File: packages/qwik-city/runtime/src/api.md [hunk 1]
```diff
@@ -167,6 +167,8 @@ export interface DocumentLink {
     type?: string;
 }
 
+type MetaNames =  "application-name" | "author" | "description" | "generator" | "keywords" | "referrer" | "theme-color" | "color-scheme" | "viewport" | "creator" | "googlebot" | "publisher" | "robots";
```
Comment (mhevery) [inline packages/qwik-city/runtime/src/api.md:170]: I don't think this is an exhaustive list Where are these defined?

Comment (mhevery) [reply packages/qwik-city/runtime/src/api.md:170]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name

I don't think there is an exhaustive list, so such a change would make this breaking to our users.

**Comment by mhevery** [comment]
So the web does not define a "standard" set of names for the `<meta>` tag see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name For this reason, I don't think we should change the type system to make it more restrictive as the user can really add any value they wish.

I am going to close this PR. Let me know if you disagree..

**Pull request closed without merging**